Easydict/Swift/Feature/Configuration/Configuration.swift

```diff
@@ -162,263 +167,246 @@ class Configuration: NSObject {
-                .sink { [weak self] _ in
-                    self?.didSetAdjustPopButtomOrigin()
-                }
-        )
+        Defaults.publisher(.adjustPopButtonOrigin)
+            .removeDuplicates()
+            .sink { [weak self] _ in
+                self?.didSetAdjustPopButtomOrigin()
+            }
+            .store(in: &cancellables)
 
-        cancellables.append(
-            Defaults.publisher(.allowCrashLog)
-                .removeDuplicates()
-                .sink { [weak self] _ in
-                    self?.didSetAllowCrashLog()
-                }
-        )
+        Defaults.publisher(.allowCrashLog)
+            .removeDuplicates()
+            .sink { [weak self] _ in
+                self?.didSetAllowCrashLog()
+            }
+            .store(in: &cancellables)
 
-        cancellables.append(
-            Defaults.publisher(.allowAnalytics)
-                .removeDuplicates()
-                .sink { [weak self] _ in
-                    self?.didSetAllowAnalytics()
-                }
-        )
+        Defaults.publisher(.allowAnalytics)
+            .removeDuplicates()
+            .sink { [weak self] _ in
+                self?.didSetAllowAnalytics()
+            }
+            .store(in: &cancellables)
 
-        cancellables.append(
-            Defaults.publisher(.clearInput)
-                .removeDuplicates()
-                .sink { [weak self] _ in
-                    self?.didSetClearInput()
-                }
-        )
+        Defaults.publisher(.clearInput)
+            .removeDuplicates()
+            .sink { [weak self] _ in
+                self?.didSetClearInput()
+            }
+            .store(in: &cancellables)
 
-        cancellables.append(
-            Defaults.publisher(.fontSizeOptionIndex)
-                .removeDuplicates()
-                .sink { [weak self] _ in
-                    self?.didSetFontSizeIndex()
-                }
-        )
+        Defaults.publisher(.fontSizeOptionIndex)
+            .removeDuplicates()
+            .sink { [weak self] _ in
+                self?.didSetFontSizeIndex()
+            }
+            .store(in: &cancellables)
 
-        cancellables.append(
-            Defaults.publisher(.appearanceType)
-                .removeDuplicates()
-                .sink { [weak self] change in
-                    let newValue = change.newValue
+        Defaults.publisher(.appearanceType)
+            .removeDuplicates()
+            .sink { [weak self] change in
+                let newValue = change.newValue
+
+                self?.didSetAppearance(newValue)
+            }
+            .store(in: &cancellables)
 
-                    self?.didSetAppearance(newValue)
+        let shortcutKeys: [Defaults.Key] = [
+            .pinShortcut,
+            .appleDictionaryShortcut,
+            .googleShortcut,
+            .eudicShortcut,
+        ]
+        for key in shortcutKeys {
+            Defaults.publisher(key)
```

Comment:
` Defaults.publisher(keys:)` can be used in this case.

Reply:
I tried it, but it doesn't seem to work with `removeDuplicates()`.

If you can fix it, please provide code suggestions.

Reply:
Yes, not having this interface https://github.com/sindresorhus/Defaults/issues/43


You can use 'options' parameter and pass empty array to it to avoid receiving initial events. But it still publish two events after value changes.

If no other solution, it's fine to approve it.

Reply:
ok，let's keep it.